pkg/controller/tf/controller.go

```diff
@@ -247,7 +256,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (res
 	if err := resourceoverrides.Handler.PreActuationTransform(&resource.Resource); err != nil {
 		return reconcile.Result{}, r.HandlePreActuationTransformFailed(ctx, &resource.Resource, fmt.Errorf("error applying pre-actuation transformation to resource '%v': %w", req.NamespacedName.String(), err))
 	}
-	requeue, err := r.sync(ctx, resource)
+
+	meta := r.provider.Meta()
+	if baseConfig, ok := meta.(*tpgconfig.Config); ok {
+		metaC := baseConfig.Clone()
+		metaC.Context = metricstransport.WithControllerName(ctx, r.controllerName) // for metrics transport
+		meta = metaC
```

Comment:
in my experience, we needed to clone this Config as the Context that it uses for HTTP requests is the one that we first configure the provider with AND NOT the ctx that we have in hand here.